### PR TITLE
Add codes for arbitary datasets

### PR DESCRIPTION
**Pull request by HelloWorldLTY**
Hi, I write a simple code for running normalization methods in arbitary single cell RNA-seq dataset as long as we have count information for such dataset. Could you please take a look at it? If you need more information or a file in better form, I can modify it. Thanks a lot.

### CODE REVIEW

**Comment by HelloWorldLTY** [comment]
Hi, I write a simple code for running normalization methods in arbitary single cell RNA-seq dataset as long as we have count information for such dataset. Could you please take a look at it? If you need more information or a file in better form, I can modify it. Thanks a lot.

File: benchmark/src/real_datast.R [hunk 1]
```diff
@@ -0,0 +1,42 @@
+source("src/transformations/transformation_helper.R")
+file = readRDS("realdataset,rds") #require both count matrix and size factor vector
```
Comment (const-ae) [inline benchmark/src/real_datast.R:2]: This looks like a typo. I assume you mean `"realdataset.rds"`, right?

File: benchmark/src/real_datast.R [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+UMI = file@assays$data$counts
+sf = file$total_counts
```
Comment (const-ae) [inline benchmark/src/real_datast.R:5]: Are these actually the total counts per cell? The size factors should be scaled such that their mean is 1. 
You can achieve this by adding:
```r
sf <- sf / mean(sf)
```

File: benchmark/src/real_datast.R [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+saveRDS(out, "real_output/sctransformp.rds")
+
+out = rand_quantile_fnc(UMI, sf, alpha)
+saveRDS(out, "real_output/rand_quantile.rds")
+
+out = logp1_fnc(UMI, sf, alpha)
+saveRDS(out, "real_output/logp1.rds")
+
+out = logp_cpm_fnc(UMI, sf, alpha)
+saveRDS(out, "real_output/logp_cpm.rds")
```
Comment (const-ae) [inline benchmark/src/real_datast.R:42]: This looks fine.

**Review by const-ae — COMMENTED** [review]
This script produces the transformed version of your "realdataset" for some of the preprocessing methods. 

To calculate the benchmark statistics like consistency or overlap with a ground truth, you will need to call for example [`calculate_10X_consistency.R`](https://github.com/const-ae/transformGamPoi-Paper/blob/master/benchmark/src/consistency_benchmark/calculate_10X_consistency.R). This script takes command line input parameters, but expects that you have already calculated the k nearest neighbors by calling [`make_knn_graph`](https://github.com/const-ae/transformGamPoi-Paper/blob/678738bb718754e84cfe6639a10520b5879fecb2/benchmark/src/consistency_benchmark/transform_consistency_data.R#L59).

File: benchmark/src/real_datast.R (hunk 1 above)
Comment (HelloWorldLTY) [reply benchmark/src/real_datast.R]: Thanks for your comments. I have fixed it.

File: benchmark/src/real_datast.R (hunk 3 above)
Comment (HelloWorldLTY) [reply benchmark/src/real_datast.R]: Thanks

File: benchmark/src/real_datast.R (hunk 2 above)
Comment (HelloWorldLTY) [reply benchmark/src/real_datast.R:5]: Thanks for your comments. I have fixed it.

But I think a more common understand for size factor or sequecing depth is the sum of total counts for per cell. See sctransform and CS-CORE.
image
image

Comment (const-ae) [reply benchmark/src/real_datast.R:5]: For the log transformation, it is important how you normalize the size factors. See the discussion around eq. (3) for more details.

Comment (HelloWorldLTY) [reply benchmark/src/real_datast.R:5]: Ok, thanks a lot. I will take a look at it and resubmit my codes.

**Comment by const-ae** [comment]
Just to clarify. I am not planning on merging any code. I am happy to use the PR interface to facilitate the discussion about your code, but you will need to run the code on your own system. 
This repository serves as a snapshot of the code that I used to generate the data for the Nature Methods paper.

**Pull request closed without merging**